src/app/services/xml-parsers/bibliography-parser.service.ts

```diff
@@ -0,0 +1,266 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { EditionDataService } from '../edition-data.service';
+
+function getStringFromTags(tagLabel: string, cit: Element) {
+  let res = '';
+  Array.from(cit.getElementsByTagName(tagLabel)).forEach(element => res += element.textContent.replace(/\s+/g, ' ') + ' ');
+
+  return res.trim();
+}
+
+function checkTwins(arr: string[], counter = 0) {
```

Comment:
What is the purpose of this function? It appears to just return `arr.some((x) => x)`